src/main/scala/diplomacy/Nodes.scala

```diff
@@ -545,15 +595,27 @@ class MixedAdapterNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
   override def description = "adapter"
   protected[diplomacy] override def flexibleArityDirection = true
   protected[diplomacy] def resolveStar(iKnown: Int, oKnown: Int, iStars: Int, oStars: Int): (Int, Int) = {
-    require (oStars + iStars <= 1, s"$context appears left of a :*= $iStars times and right of a :=* $oStars times; at most once is allowed")
+    require (oStars + iStars <= 1,
+      s"""$context
```

Comment:
As above, change the statement order for these and preface with "Diplomacy has detected a problem with your graph:"